app/src/main/java/com/babylon/wallet/android/presentation/onboarding/restore/mnemonics/RestoreMnemonicsViewModel.kt

```diff
@@ -128,10 +130,28 @@ class RestoreMnemonicsViewModel @Inject constructor(
         }
     }
 
-    fun onSkipClick() {
+    fun onSkipSeedPhraseClick() {
         viewModelScope.launch { showNextRecoverableFactorSourceOrFinish() }
     }
 
+    fun onSkipMainSeedPhraseClick() {
+        _state.update {
+            it.copy(screenType = State.ScreenType.NoMainSeedPhrase)
+        }
+    }
+
+    fun skipMainSeedPhraseAndCreateNew() {
```

Comment:
if I am not mistaken you always create a new main seed phrase no matter how many times we skip the previous seed phrases, right?

Wouldn't be sufficient to create this new main seed phrase only in the first skip so we dont have to use biometric every time we skip a seed phrase?
Unless this is a requirement!

Reply:
> if I am not mistaken you always create a new main seed phrase no matter how many times we skip the previous seed phrases, right?
> 
> Wouldn't be sufficient to create this new main seed phrase only in the first skip so we dont have to use biometric every time we skip a seed phrase? Unless this is a requirement!

So this is first skip (main seed phrase) and it is being used only here for the first time. When you skip other seed phrases this is not called. Is it what you had in mind ?

Reply:
@giannis-rdx Pushed fix for this

Reply:
just tested and when I click the "skip this seed phrase for now" it goes back to the onboarding screen